tests/beaker_tests/lib/utilitylib.rb

```diff
@@ -374,3 +382,131 @@ def bgp_title_pattern_munge(tests, id, provider=nil)
+    break if vdc_allocate_interface_get(vdc, intf)
+    logger.info("'#{intf}' is not allocated to VDC, allocate it now...")
+    vdc_allocate_interface_set(vdc, intf)
+    break if vdc_allocate_interface_get(vdc, intf)
+    prereq_skip(testheader, testcase,
+                "Unable to allocate interface '#{intf}' to VDC")
+  end
+
+  step "Add switchport config to #{intf}" do
+    cmd = get_vshell_cmd("conf t ; int #{intf} ; #{tests[:config_switchport]}")
+    on(agent, cmd, pty: true)
+  end if tests[:config_switchport]
+
+  step 'Add bridge-domain global config' do
+    cmd = get_vshell_cmd("conf t ; #{tests[:config_bridge_domain]}")
+    on(agent, cmd, pty: true)
+  end if tests[:config_bridge_domain]
+end
+
+# Helper to raise skip when prereqs are not met
+def prereq_skip(testheader, testcase, message)
+  logger.error("** PLATFORM PREREQUISITE NOT MET: #{message}")
+  raise_skip_exception(testheader, testcase)
+end
+
+# Return an interface name from the first MT-full compatible line module found
+def mt_full_interface
+  # Search for F3 card on device, create an interface name if found
+  cmd = get_vshell_cmd('sh mod')
+  out = on(agent, cmd, pty: true).stdout[/^(\d+)\s.*N7K-F3/]
+  slot = out.nil? ? nil : Regexp.last_match[1]
+  "ethernet#{slot}/1" unless slot.nil?
+end
+
+# Return the default vdc name
+def default_vdc_name
+  cmd = get_vshell_cmd('sh run vdc')
+  out = on(agent, cmd, pty: true).stdout[/^vdc (\S+) id 1$/]
+  out.nil? ? nil : Regexp.last_match[1]
+end
+
+# Check for presence of limit-resource module-type
+def limit_resource_module_type_get(vdc, mod)
+  cmd = get_vshell_cmd("sh vdc #{vdc} detail")
+  pat = Regexp.new("vdc supported linecards:.*(#{mod})")
+  out = on(agent, cmd, pty: true).stdout.match(pat)
+  out.nil? ? nil : Regexp.last_match[1]
+end
+
+# Set limit-resource module-type
+def limit_resource_module_type_set(vdc, mod)
+  # Turn off prompting
+  cmd = get_vshell_cmd('terminal dont-ask persist')
+  on(agent, cmd, pty: true)
+
+  cmd = get_vshell_cmd("conf t ; vdc #{vdc} ; "\
+                       "limit-resource module-type #{mod}")
+  on(agent, cmd, pty: true)
+
+  # Reset dont-ask to default setting
+  cmd = get_vshell_cmd('no terminal dont-ask persist')
+  on(agent, cmd, pty: true)
+end
+
+# Check for presence of interface in vdc allocated interfaces
+def vdc_allocate_interface_get(vdc, intf)
+  intf_pat = "(#{intf}) " # note trailing space
+  cmd = get_vshell_cmd("sh vdc #{vdc} membership")
+  out = on(agent, cmd, pty: true).stdout.match(
+    Regexp.new(intf_pat, Regexp::IGNORECASE))
+  out.nil? ? nil : Regexp.last_match[1]
+end
+
+# Add interface to vdc's allocated interfaces
+def vdc_allocate_interface_set(vdc, intf)
+  # Turn off prompting
+  cmd = get_vshell_cmd('terminal dont-ask persist')
+  on(agent, cmd, pty: true)
+
+  cmd = get_vshell_cmd("conf t ; vdc #{vdc} ; "\
+                       "allocate interface #{intf}")
+  on(agent, cmd, pty: true)
```

Comment:
@glennmatthews Our long term plan is to remove all instances of vshell for validating 'show output'.  Are you also suggesting we remove config cases?  Do you envision we would replace it with similar config api's that we have in minitest?